src/meshes/p4est_mesh.jl

```diff
@@ -469,12 +474,186 @@ function p4est_mesh_from_standard_abaqus(meshfile, mapping, polydeg,
+                        # Front face is position 3 in p4est indexing
+                        boundary_names[3, tree] = boundary
+                    end
+                    # Check "back face" (y_max)
+                    if tree_nodes[3] in node_set_dict[boundary] &&
+                       tree_nodes[4] in node_set_dict[boundary] &&
+                       tree_nodes[7] in node_set_dict[boundary] &&
+                       tree_nodes[8] in node_set_dict[boundary]
+                        # Front face is position 4 in p4est indexing
+                        boundary_names[4, tree] = boundary
+                    end
+                    # Check "left face" (x_min)
+                    if tree_nodes[1] in node_set_dict[boundary] &&
+                       tree_nodes[4] in node_set_dict[boundary] &&
+                       tree_nodes[5] in node_set_dict[boundary] &&
+                       tree_nodes[8] in node_set_dict[boundary]
+                        # Left face is position 1 in p4est indexing
+                        boundary_names[1, tree] = boundary
+                    end
+                    # Check "right face" (x_max)
+                    if tree_nodes[2] in node_set_dict[boundary] &&
+                       tree_nodes[3] in node_set_dict[boundary] &&
+                       tree_nodes[6] in node_set_dict[boundary] &&
+                       tree_nodes[7] in node_set_dict[boundary]
+                        # Right face is position 2 in p4est indexing
+                        boundary_names[2, tree] = boundary
+                    end
+                    # Check "bottom face" (z_min)
+                    if tree_nodes[1] in node_set_dict[boundary] &&
+                       tree_nodes[2] in node_set_dict[boundary] &&
+                       tree_nodes[3] in node_set_dict[boundary] &&
+                       tree_nodes[4] in node_set_dict[boundary]
+                        # Bottom face is position 5 in p4est indexing
+                        boundary_names[5, tree] = boundary
+                    end
+                    # Check "top face" (z_max)
+                    if tree_nodes[5] in node_set_dict[boundary] &&
+                       tree_nodes[6] in node_set_dict[boundary] &&
+                       tree_nodes[7] in node_set_dict[boundary] &&
+                       tree_nodes[8] in node_set_dict[boundary]
+                        # Top face is position 6 in p4est indexing
+                        boundary_names[6, tree] = boundary
+                    end
+                end
+            end
+        end
+    end
 
     return p4est, tree_node_coordinates, nodes, boundary_names
 end
 
+function parse_elements(meshfile, n_trees, n_dims)
+    @assert n_dims ∈ [2, 3] "Only 2D and 3D meshes are supported"
+    element_types = n_dims == 2 ?
+                    [
+        "*ELEMENT, type=CPS4",
+        "*ELEMENT, type=C2D4",
+        "*ELEMENT, type=S4",
+    ] :
+                    ["*ELEMENT, type=C3D8"]
```

Comment:
The formatting of this looks weird. Also, are there no other 3D `ELEMENT` types that gmsh will spit out in valid ABAQUS mesh files?

Reply:
I am also not really happy with the format, I will check if this can be improved - maybe at the expense of exchanging the conditional assignment operator `?` for an `if else` construct.

Indeed, `p4est` can only handle `C3D8` in Abaqus format, see page 11 of the [documentation](https://p4est.github.io/p4est-howto.pdf).